base_layer/wallet/src/output_manager_service/storage/sqlite_db/output_sql.rs

```diff
@@ -243,6 +243,36 @@ impl OutputSql {
                     ),
                 };
             },
+
+            UtxoSelectionFilter::MustInclude { commitments } => {
+                if commitments.is_empty() {
+                    // If no commitments specified, fall back to standard behavior
+                    query = query.filter(
+                        outputs::output_type
+                            .eq(i32::from(OutputType::Standard.as_byte()))
+                            .or(outputs::output_type.eq(i32::from(OutputType::Coinbase.as_byte()))),
+                    );
+
+                    if selection_criteria.excluding_onesided {
+                        query = query.filter(outputs::source.ne(OutputSource::OneSided as i32));
+                    }
```

Comment:
Suggested for improved maintenance
```suggestion
                let mut selection_criteria = selection_criteria.clone();
                selection_criteria.filter = UtxoSelectionFilter::Standard;
                return OutputSql::fetch_unspent_outputs_for_spending(
                    &selection_criteria,
                    amount,
                    tip_height,
                    conn,
                );
```

Reply:
Implemented suggested code.